spring-kafka/src/test/java/org/springframework/kafka/listener/adapter/FilteringAdapterTests.java

```diff
@@ -72,4 +75,173 @@ public void testBatchFilterAckDiscard() throws Exception {
 		verify(listener, never()).onMessage(any(List.class), any(Acknowledgment.class));
 	}
 
+	@Test
+	public void listener_should_not_be_invoked_on_emptyList_and_ignoreEmptyBatch_true() throws Exception {
+		// Given :
+		final RecordFilterStrategy<String, String> filter = new RecordFilterStrategy<>() {
+			@Override
+			public boolean filter(ConsumerRecord<String, String> consumerRecord) {
+				return true;
+			}
+
+			@Override
+			public List<ConsumerRecord<String, String>> filterBatch(
+					List<ConsumerRecord<String, String>> consumerRecords) {
+				// SUT
+				return List.of();
+			}
+
+			@Override
+			public boolean ignoreEmptyBatch() {
+				// SUT
```

Comment:
What is `SUT`, please?

I'm not sure that we need obvious comments in tests at all.

Reply:
I agree! 
I removed all of them from test.

FYI, `SUT` is `System Under Test`.